pkg/config/lima_config_applier.go

```diff
@@ -80,45 +96,62 @@ func NewLimaApplier(
 	cfg *Finch,
 	cmdCreator command.Creator,
 	fs afero.Fs,
-	limaConfigPath string,
+	limaDefaultConfigPath string,
+	limaOverrideConfigPath string,
 	systemDeps LimaConfigApplierSystemDeps,
 ) LimaConfigApplier {
 	return &limaConfigApplier{
-		cfg:            cfg,
-		cmdCreator:     cmdCreator,
-		fs:             fs,
-		limaConfigPath: limaConfigPath,
-		systemDeps:     systemDeps,
+		cfg:                    cfg,
+		cmdCreator:             cmdCreator,
+		fs:                     fs,
+		limaDefaultConfigPath:  limaDefaultConfigPath,
+		limaOverrideConfigPath: limaOverrideConfigPath,
+		systemDeps:             systemDeps,
 	}
 }
 
 // Apply writes Lima-specific config values from Finch's config to the supplied lima config file path.
 // Apply will create a lima config file at the path if it does not exist.
-func (lca *limaConfigApplier) Apply(isInit bool) error {
-	if cfgExists, err := afero.Exists(lca.fs, lca.limaConfigPath); err != nil {
-		return fmt.Errorf("error checking if file at path %s exists, error: %w", lca.limaConfigPath, err)
-	} else if !cfgExists {
-		if err := afero.WriteFile(lca.fs, lca.limaConfigPath, []byte(""), 0o600); err != nil {
-			return fmt.Errorf("failed to create the an empty lima config file: %w", err)
-		}
+func (lca *limaConfigApplier) ApplyInit() error {
+	if err := afero.WriteFile(lca.fs, lca.limaDefaultConfigPath, []byte(""), 0o600); err != nil {
+		return fmt.Errorf("failed to create the an empty lima config file: %w", err)
+	}
+
+	var limaCfg limayaml.LimaYAML
+
+	if limaCfg.Rosetta.Enabled == nil {
+		limaCfg.Rosetta.Enabled = pointer.Bool(false)
+		limaCfg.Rosetta.BinFmt = pointer.Bool(false)
 	}
 
-	b, err := afero.ReadFile(lca.fs, lca.limaConfigPath)
+	cfgAfterInit, err := lca.applyInit(&limaCfg)
 	if err != nil {
-		return fmt.Errorf("failed to load the lima config file: %w", err)
+		return fmt.Errorf("failed to apply init-only config values: %w", err)
 	}
+	limaCfg = *cfgAfterInit
 
-	var limaCfg limayaml.LimaYAML
-	if err := yaml.Unmarshal(b, &limaCfg); err != nil {
-		return fmt.Errorf("failed to unmarshal the lima config file: %w", err)
+	limaCfgBytes, err := yaml.Marshal(limaCfg)
+	if err != nil {
+		return fmt.Errorf("failed to marshal the lima config file: %w", err)
+	}
+
+	if err := afero.WriteFile(lca.fs, lca.limaDefaultConfigPath, limaCfgBytes, 0o600); err != nil {
+		return fmt.Errorf("failed to write to the lima config file: %w", err)
 	}
 
-	// Unmarshall with custom unmarshaler for Disk:
-	// https://github.com/lima-vm/lima/blob/v0.17.2/pkg/limayaml/load.go#L16
-	if err := goyaml.UnmarshalWithOptions(b, &limaCfg, goyaml.DisallowDuplicateKey(),
-		goyaml.CustomUnmarshaler[limayaml.Disk](unmarshalDisk)); err != nil {
-		return fmt.Errorf("failed to unmarshal the lima config file: %w", err)
+	err = lca.Apply()
```

Comment:
put in a different function rather than Apply and call both function, making it clear what each part does.